test/public/defaults.js

```diff
@@ -510,11 +510,19 @@ module.exports.checkMismatchingUrlParam = async (page, expectedUrlParameters) =>
  * @return {Promise<void>} resolve once column values were checked
  */
 module.exports.expectColumnValues = async (page, columnId, expectedInnerTextValues) => {
-    await page.waitForFunction((columnId, expectedInnerTextValues) => {
-        // Browser context, be careful when modifying
-        const names = [...document.querySelectorAll(`table tbody .column-${columnId}`)].map(({ innerText }) => innerText);
-        return JSON.stringify(names) === JSON.stringify(expectedInnerTextValues);
-    }, { timeout: 1500 }, columnId, expectedInnerTextValues);
+    const timeout = 1500;
+    try {
+        await page.waitForFunction((columnId, expectedInnerTextValues) => {
+            // Browser context, be careful when modifying
+            const names = [...document.querySelectorAll(`table tbody .column-${columnId}`)].map(({ innerText }) => innerText);
+            return JSON.stringify(names) === JSON.stringify(expectedInnerTextValues);
+        }, { timeout }, columnId, expectedInnerTextValues);
+    } catch {
+        throw new Error(`Column (${columnId}) is has incorrect values after timeout (${timeout}):
+            expected: [${expectedInnerTextValues}],
+            actual:   [${await this.getColumnCellsInnerTexts(page, columnId)}]
+        `);
+    }
```

Comment:
The idea is very good, but I would rather not use `getColumnCellsInnerTexts`, you should get the value, compare it, then throw if they do not match using the value found in the first place. If you do it in two steps, the value can change in the mean time and provide a misleading error message

Reply:
`getColumnCellsInnerTexts` cannot be called inside `waitForFunciton` and also cannot pass it's result to it.
Handle case when column has expected values only after timeout